### ISSUE TITLE
File mode check doesn't look at File references

### ISSUE THREAD

**Issue by halfninja** [rodjek/puppet-lint#796]
It should warn in both cases here but it only looks at `file` resources.
```
# no warning reported
File {
  mode => 0666,
}

# reports warning
file { "/tmp/name" :
  mode => 0666,
}
```
